src/clarity-angular/emphasis/alert/providers/icon-and-types-service.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Injectable } from "@angular/core";
+
+@Injectable()
+export class AlertIconAndTypesService {
+    private defaultIconShape = "info-circle";
+    private _alertIconShape = "";
+    private _alertType = "info";
+
+    get alertType(): string {
+        return this._alertType;
+    }
+    set alertType(val: string) {
+        if (this.alertTypes.indexOf(val) > -1) {
+            this._alertType = val;
+        }
+    }
+
+    // alert-* types are deprecated!
+    private alertTypes: string[] = [
+        "alert-info", "alert-warning", "alert-danger", "alert-success",
+        "info", "warning", "danger", "success"
+    ];
```

Comment:
Please make this a constant at the module level, right now you're instantiating a different array for every single alert in the app.